core/network/corda-interop-app/interop-workflows/src/main/kotlin/com/weaver/corda/app/interop/flows/ViewVerification.kt

```diff
@@ -186,22 +186,42 @@ fun verifyFabricNotarization(
         println("Fabric view data: $fabricViewData\n")
 
         // 2. Verify address in payload is the same as original address
-        val interopPayload = InteropPayloadOuterClass.InteropPayload.parseFrom(fabricViewData.response.payload)
+        /*val interopPayload = InteropPayloadOuterClass.InteropPayload.parseFrom(fabricViewData.response.payload)
         println("Interop payload: $interopPayload")
         if (interopPayload.address != addressString) {
             println("Address in response does not match original address: Original: $addressString Response: ${interopPayload.address}")
             return Left(Error("Address in response does not match original address: Original: $addressString Response: ${interopPayload.address}"))
+        }*/
+
+        // TODO: Handle encrypted (confidential) payloads in Fabric views
+        var responsePayload = ""
+        var responseIndex = 0
+        for (endorsedProposalResponse in fabricViewData.endorsedProposalResponsesList) {
+            val chaincodeAction = ProposalPackage.ChaincodeAction.parseFrom(endorsedProposalResponse.payload.extension)
+            val interopPayload = InteropPayloadOuterClass.InteropPayload.parseFrom(chaincodeAction.response.payload)
+            println("Interop payload: $interopPayload")
+            if (responseIndex == 0) {
+                responsePayload = chaincodeAction.response.payload.toString()
+            } else if (responsePayload != chaincodeAction.response.payload.toString()) {
+                println("Mismatching payloads in proposal responses: 0 - $responsePayload,  $responseIndex: ${chaincodeAction.response.payload}")
+                return Left(Error("Address in response does not match original address: Original: $addressString Response: ${interopPayload.address}"))
```

Comment:
This if block is about comparing `payload`, but while throwing Error it says `address`.
Also not sure if printing payload bytes in simple string would help? could print in `base64` or `utf-8`?

Reply:
Can you suggest an alternative call here? Kotlin syntax isn't my strong suite.

Reply:
you can use `responsePayload = Base64.getEncoder().encodeToString(chaincodeAction.response.payload)`, to convert to base64 string